fxfs/fxfs_test.go

```diff
@@ -0,0 +1,198 @@
+package fxfs
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	fxfscore "github.com/functionland/go-fula/fxfs/core/api"
+	fxiface "github.com/functionland/go-fula/fxfs/core/iface"
+	pfs "github.com/functionland/go-fula/fxfs/core/pfs"
+	"github.com/ipfs/go-datastore"
+	syncds "github.com/ipfs/go-datastore/sync"
+	"github.com/ipfs/go-filestore"
+	files "github.com/ipfs/go-ipfs-files"
+	keystore "github.com/ipfs/go-ipfs-keystore"
+	"github.com/ipfs/interface-go-ipfs-core/path"
+	"github.com/ipfs/kubo/config"
+	"github.com/ipfs/kubo/core"
+	"github.com/ipfs/kubo/core/bootstrap"
+	"github.com/ipfs/kubo/core/coreapi"
+	mock "github.com/ipfs/kubo/core/mock"
+	"github.com/ipfs/kubo/core/node/libp2p"
+	"github.com/ipfs/kubo/repo"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+)
+
+const testPeerID = "QmTFauExutTsy4XP6JbMFcw2Wa9645HJt2bTqL6qYDCKfe"
```

Comment:
We probably should not encourage peer IDs that don't include public key

Reply:
I'm not sure what you mean by this @masih. Are you saying that we should use some other type of peerID for tests or are you talking about reconsidering the type of peerID entirely?

Reply:
Sorry I should have been clearer. There are different kinds of peer ID. The kind here is a CID, which doesn't implicitly include public key. There are other kinds of peer ID that do. The advantage of including public key is that we won't need an extra hop to get it; if we know the peer ID we know the public key. 

For testing purposes using a CID kind is fine 👍 For the running system we probably should use the other kind 🙂

Reply:
Thank you @masih for explaining. 
Can you point me to a reference or a piece of code for this type of peerID you are talking about? I can see that one can verify if a peerID is derived from a certain public key, but obtaining the public key from the peerID (which I believe is a hash built from the public key) is what I can't find anything about it.

Reply:
Take look here: https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md

Reply:
According to the spec, we can obtain the public key from the peer-id when the serialized public key has a length of fewer than 42 bytes. 

> If the length of the serialized bytes is less than or equal to 42, compute the "identity" multihash of the serialized bytes. In other words, no hashing is performed, but the multihash format is still followed (byte plus varint plus serialized bytes).

We have to use Ed25519 keys to limit the size, therefore being able to obtain the public key from the peer-id. I think we are currently using RSA keys, @Ruffiano can clarify on this.

Also, according to the spec, we can assume that all of the client peers support the Ed25519 public and private keys. But inside the Blox network, we have to support RSA for IPFS DHT. 

> Implementations MUST support Ed25519. Implementations SHOULD support RSA if they wish to interoperate with the mainline IPFS DHT and the default IPFS bootstrap nodes. Implementations MAY support Secp256k1 and ECDSA, but nodes using those keys may not be able to connect to all other nodes.

@masih how about resolving this conversation here? I can create a discussion for this topic, I think we have more to talk about before applying any changes to the code.